substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1920,6 +1922,47 @@ pub mod pallet {
 
 			Ok(Pays::No.into())
 		}
+
+		/// Updates a batch of controller accounts to their corresponding stash account if they are
+		/// not the same. Ignores any controller accounts that do not exist, and does not operate if
+		/// the stash and controller are already the same.
+		///
+		/// Effects will be felt instantly (as soon as this function is completed successfully).
+		///
+		/// The dispatch origin must be Root.
+		#[pallet::call_index(28)]
+		#[pallet::weight(T::WeightInfo::deprecate_controller_batch(controllers.len() as u32))]
+		pub fn deprecate_controller_batch(
+			origin: OriginFor<T>,
+			controllers: BoundedVec<T::AccountId, T::MaxControllersInDeprecationBatch>,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
```

Comment:
Using StakingAdmin origin?
```suggestion
			T::AdminOrigin::ensure_origin(origin)?;
```